sphinx/templates/latex/tabular.tex_t

```diff
@@ -1,3 +1,11 @@
+<%- if table.align -%>
+\begin{table}[htb]
+  <%- if table.align == 'center' -%>
+    \centering
+  <%- else -%>
+    \begin{flush<%= table.align%>}
+  <%- endif %>
+<% endif -%>
 <%- if table.caption -%>
 \begin{threeparttable}
```

Comment:
The advantage of `threeparttable` is that it handles the caption automatically. But its notion of `tablenotes` causes it to patch the `tabular` or `tabulary`, hence if one wants to use `tabu` one would have to try to imitate which is low-level LaTeX hacking. If we don't use `threeparttable` we can emulate without too much difficulty the treatment of the caption (already Sphinx loads `capt-of` package which facilitates use of `\caption` outside of reguler `table` environment). In fact `threeparttable` wraps the caption to the natural width of the tabular, which is not so interesting if text is not allowed to flow around it. In conclusion, dropping `threeparttable` does not mean we have to switch to `table`. We can do it by our own means without using regular LaTeX floats. I don't think I have time today, but I will look tomorrow.

There have been some requests (I lost reference) on Sphinx issue tracker to let the table floats in PDF output. I think this can indeed be made a new config option. But `longtable` is not a float and (not tested, but I expect that) it can not be wrapped in `table`. In fact a LaTeX `table` is an object which at some point should appear without any page break on one page. `longtable` to the contrary allows page breaks. It can thus not float in the LaTeX technical sense. There is no "float" in LaTeX which is multi-page. Thus if we start having use of `table` for smaller tables we may end up in situation that sometimes for some mysteriousreason the small table will be printed 3 pages later after the very long longtable coming next. This means it is important that H specifier is available if we do that.

Sorry for very long explanation only to say that:

- we can handle captions after having dropped `threeparttable` and without switching to `table`,

- we can switch to `table`  but `longtable` will remain as is,

- the switch to `table` should possibly be made a config option, 

  * and if True, then user should be able to use H specifier in some cases but not necessarily all the time in the document,

  * if user does not say to make table float, it is like current situation, or like saying table floats but use H specifier all the time; hence I think it should not be a possibility for user to say to use table floats with H specifier all the time, in such case, the user must use default behaviour that tables are not (LaTeX) floats.